test/staged.jl

```diff
@@ -170,7 +170,7 @@ let gf_err, tsk = @async nothing # create a Task for yield to try to run
     end
     @test_throws ErrorException gf_err()
     @test_throws ErrorException gf_err()
-    @test gf_err_ref[] == 2
+    @test gf_err_ref[] == 4
```

Comment:
It used to fail on construction (while we were holding the method locks), which was a bad idea. Now it fails once when trying to type-infer it, and again when it tries to run it (for each of the two attempted calls)